packages/nocodb/src/lib/meta/api/columnApis.ts

```diff
@@ -50,6 +50,16 @@ export enum Altered {
   UPDATE_COLUMN = 8,
 }
 
+// generate unique foreign key constraint name for foreign key
+const generateFkConstraintName = (parent: TableType, child: TableType) => {
+  const constraintName = `fk_${parent.table_name
+    .replace(/\W+/g, '_')
+    .slice(0, 10)}_${child.table_name
+    .replace(/\W+/g, '_')
+    .slice(0, 10)}_${randomID(15)}`;
+  return constraintName;
+};
+
```

Comment:
I'd suggest to add some comments to explain why we slice the table name as i think it's not quite intuitive at the first sight. For example, we may mention sth like in order to prevent FK key name length from exceeding the max length during creation.